File: code/java/src/main/java/state/TrafficSystem.java
package state;

class TrafficSystem {
    public static void main(String[] args) {
        TrafficLight light = new TrafficLight(new GreenState());

        // Green
        light.perform();

        // Green to Yellow
        light.setState(new YellowState());
        light.perform();

        // Yellow to Red
        light.setState(new RedState());
        light.perform();
    }
}

File: code/python/state/TrafficSystem.py
from TrafficLight import TrafficLight
from GreenState import GreenState
from YellowState import YellowState
from RedState import RedState

if __name__ == "__main__":
    light = TrafficLight(GreenState())

    # Green
    light.perform()

    # Green to Yellow
    light.set_state(YellowState())
    light.perform()

    # Yellow to Red
    light.set_state(RedState())
    light.perform()
